### ISSUE TITLE
Pirates (and similar) running from unwinnable fights

### ISSUE THREAD

**Issue by AlbertNewton** [endless-sky/endless-sky#3898]
When going between Human and Wanderer space, which I do a lot, I end up going through the Deep's anarchist systems, and the nearby uninhabited systems. When this happens, I often encounter tiny pirate fleets which I could destroy in about one second, if I didn't jump away first. These lone Sparrows seem to think they can destroy me, even though I have an enormously powerful fleet, and I think it would be better for ships that know they cannot win a fight to flee. Of course, Republic ships and similar should not flee, because they are under orders, and maybe Korath could be more optimistic, overestimating themselves, while the Kor Sestor and Mereti would not flee.

**Comment by tehhowch** [comment]
This is already (kinda) true. Ships consider strength based on cost, and further, they consider nearby strength. If your fleet is far away, or you use cheat weapons that cost nothing but have high damage, AI ships won't consider your ship a threat, especially if they have allies nearby. However, to flee they generally need to already be pretty hurt, or you have to be far enough away. If you're jumping in, they're probably within the attack threshold already.

The personality used to instantiate the ship matters, as does the source of the ship - many mission ships (which does include hostile pirates) are configured such that they will follow you (ensuring conflict) and won't flee.

**Comment by DingusShingleton** [comment]
Idk... If I'm in a sparrow I don't need to take a single damage from a derecho to know that thing is gonna infinity glove me into non existence

**Comment by Pointedstick** [comment]
Yeah, I will agree that ships standing no chance don't retreat often enough. If they're outside the invisible zone of engagement, they won't go towards you, but if they're already within it, their priority seems to be fighting over leaving it, which IMHO is wrong and leads to this suicidal behavior.

**Comment by DingusShingleton** [comment]
"You can't win... But there are alternatives to fighting."

     – Obi Wan "Old Ben" Kenobi

Could such pirates attempt to bribe you into letting them live?

**Comment by Amazinite** [comment]
Here’s an idea: 

* Alter the random NPC AI to flee significantly sooner if they’re clearly going to lose from the start. (A Sparrow shouldn’t need to take damage from my Derecho before realizing it isn’t going to win, especially as it’ll be long dead by the time it even thinks of trying to flee.)
* Alter the player escort AI to not fire unless targeted or told to fire

These changes will make it so that upon entering a system with a much larger fleet than the pirates in the system, any pirates will not target the player ships and instead begin to flee from the player while player escorts won’t try and slaughter the fleeing ships unless the player specifically tells them to do so. To the pirates, it’s like “oh crap, I need to get out of here if I want to live.” To the player escorts, it’s like “they’re already fleeing from us, so there’s no point in even wasting the energy to destroy them.”

**Comment by ghoulavenger** [comment]
I'm not against this type of change per say, but I have to wonder about it. Personally I think the fleet attraction system as of 0.9.8 (I'm using the latest steam version) is a stronng counter to this already. I attract pirate fleets with my massive fleet in human space, because I'm just hauling that much cargo, the pirate fleets are always instantly annihilated and pose no significant threat (like 70+ hurricanes with over 100 well armed deep rivers, offhand without checking for accuracy) -- exception is when I get ahead of the fleet and they all target the flagship, but that doesn't happen very often. Which means if pirates always fled if they knew they were going to lose, wouldn't they just not show up in the first place? 

Second point worthy of considering is that farming pirates for money is a thing, and to do so losslessly you want a significant tactical advantage. If pirates didn't fight back in this instance or constantly ran, this would either be unprofitable (got away too quickly) OR extremely easy (since they wouldn't necessarily shoot back -- guns only fire forward and turrets need time to turn -- so there would be less lost to crossfire).

**Comment by DingusShingleton** [comment]
No offense, but balancing the game for people with 10+ billion credit fleet is very pointless.

The second point, though, has merit, but I think if the parameters of when pirates chose to run away were balanced properly, then it wouldn't impact too much. They should stay and fight even if they know they will lose, but if the odds get stacked to the point of absurdity I think it would actually be a bit gratifying to see measly sparrows and wasps and such fleeing in abject terror. And by that point, it's not even worth my time to capture those ships.

**Comment by Amazinite** [comment]
>Which means if pirates always fled if they knew they were going to lose, wouldn't they just not show up in the first place?

Which is why I mentioned how this could possibly be a new personality instead of a complete removal of the current system; pirate raids wouldn't have said personality.

**Comment by ghoulavenger** [comment]
@DingusShingleton my point was not that the game was unbalanced or that it should be balanced around my massive fleet. It was that the game itself sets up unwinnable fights for pirates by virtue of its own systems. Those particular battles CAN and have been avoided by simply TAKING LESS deep rivers with me. If pirates want to suicide because I have a few more freighters in my fleet I don't see a problem.

As to the second comment I have no strong feeling either way. They still add to the mess of battle, but capturing them is just as likely to end up with them destroyed, so I often either ignore them or destroy them deliberately.

@Amazinite so your suggestion is to basically have some suicidal pirates and some non-suicidal pirates? Well it might work.

**Comment by esksp** [comment]
In one of my games I own by now a couple heavy warships, heavily armed, optimized by alien technology (found about all the aliens out there and earned their licenses). To make it short, nothing in human space could make me even worry.
And still the small pirate interceptors keep attacking me on sight, again and again, and I destroy them instantly, while they're closing in, one after the other. It's aggravating. It's boring.

Shouldn't absurdly weaker enemies avoid you, flee, or at least stop attacking you and try to save their hides? Are they all religious fanatics? Suicidal madmen?
Yes, I know, the "heroic" personality. But outside of specific cases it is a little silly, especially for pirates, so could we have a less suicidal but still aggressive personality?

A personality which is **just as aggressive as the "heroic"**, but which **will immediately turn and flee if**:
1. Your **combat rating** is off the charts compared to his
2. Your **cumulative firepower** (you+your escorts) is more than 1.5 times his (him+his friends)
3. His **shields** OR **hull** OR **energy** have slipped under 50%

- So there would be cases where when seeing you, the enemies will just make a hasty U-turn and run for the hills: They have decided they're definitely out of their league, and wish to live to tell their grand-children about that encounter (case 1).
- Or where the enemies initially fight, but after you have destroyed a couple of them, the rest suddenly just turns and flees (you have triggered case 2).
- Or a lightly damaged ship flees, but after a while, energy and shields recharged, it turns and attacks again (case 3). BTW, that "fleeing" ship counts towards case 2, its firepower isn't available anymore.

**When fleeing**, enemies would flee to the **outer reaches of the current system** (would need to become `unconstrained` while actively fleeing). This is so you can still try to capture them.
But it won't be easy, because **if you hunt a fleeing ship down and damage it** (further) by firing at them for over 2-3 seconds, **they will try to jump away** from that system. You'll have to be quick.

Such a behavior would give the enemies a more human feeling, less like mosquitoes. 
The drawback is that at some point you will become too powerful for pirates or such to even dare cross your path. But well, at that point you most likely have better things to do than to chase puny pirates for a couple dollars, so you won't miss the incessant buzzing too much... 

The only real **issue** I see is calculating that "firepower". I don't know what's under the hood, so it might or it might not be easy. 


(Initially posted as #10647)

**Comment by Anarchist2** [comment]
Case 3 is already in the game. If a ship has less than 25% of its health before it becomes disabled, it attempts to jump away. This increases to 50% for ships with the `coward` personality, though that personality doesn't show up on pirate fleets. The easiest way to see this in-game is in Wah Ki; Unfettered ships are rarely disabled or destroyed, because their fleeing AI kicks in well before then.

Case 2 is in the game, sorta. As long as you're beyond a certain range, enemy ships that are much weaker than you won't intentionally approach and attack you. Once you do go within range, however, they won't stop until you fly back out of range or they lose enough health to go into fleeing mode. As pointed out by https://github.com/endless-sky/endless-sky/issues/3898#issuecomment-413110027, this is the main reason for why pirates seem so suicidal, despite the existence of these measures.

**Comment by esksp** [comment]
> Case 3 is already in the game. [...]

Indeed, I noticed this, but it rarely kicks in (except in the case you mentioned) because once under 25% health the most common enemy ships (including but not limited to pirates) are dead in a matter of seconds. Given the time it takes the AI to prepare a jump, they are usually destroyed before getting a chance to jump away.
I think this might just need some adjustment (maybe raise the threshold to 30%? Don't know, this is a question of trial & error till you find a value which feels right).


> Case 2 is in the game, sorta. As long as you're beyond a certain range, enemy ships that are much weaker than you won't intentionally approach and attack you.

Yes, but the range thing only applies _very_ rarely (if at all), because ships are not randomly distributed over the whole system, they tend to flock around the arrival/depart points, which more often than not happen to also be near the inhabited planets. In short, in every system there are fixed choke points where most/all ships gather or fly through at some point. I actually never noticed a ship hesitating to attack me. Usually they come jumping in and immediately, invariably charge all guns blazing--to their death...

Besides, if you think about it, this rule doesn't make _any_ sense: Seriously, if a ship _is_ weaker than you, will proximity make it strong enough to have a chance? No, not at all. On the contrary, their drive to flee no matter what should increase as distance diminishes. They might slow down if there is enough distance, but if you start getting closer they should _run_.

**Comment by Mernom** [comment]
I think that the systems in human space being so small contributes to this. There are less places for pirate ships to be that isn't close enough for them to enter forced engagement range.
